### ISSUE TITLE
Create a query builder

### ISSUE THREAD

**Issue by maruiz93** [kubearchive/kubearchive#649]
We are currently exposing the complete SQL statement for each query but this is not very maintainable when we start adding filters that are applicable to several queries (like labels or annotations)

### Acceptance Criteria

* Separate the SQL statements for `SELECT` queries into selector and filters and create a builder to build the query with the different pieces

**Comment by ggallen** [comment]
There are SQL query builder package out there.  I think we should use one of those rather than trying to write our own.

Here's one: https://pkg.go.dev/github.com/huandu/go-sqlbuilder

**Comment by maruiz93** [comment]
I think that those libraries are too opinionated for KubeArchive's usecase.
My intention was to have the queries separated in:
- One selector, meaning the `SELECT` + `FROM` + `JOIN` clause because it doesn't make sense to me having it separate, a specific select is going to be always with the same `FROM` (e.g., `SELECT url FROM log_url`) and I believe that the experience of providing them together for the different drivers is going to be better.
- None, one or more filters, meaning `WHERE` clauses used with `AND`.
- None or one `ORDER` clause
- None or one `LIMIT` clause

For example the `go-sqlbuilder` forces you to have the `SELECT` and `FROM` clauses separated, and it provides a lot of condition methods that I believe we don't need as we rely a lot on the specific JSON functionalities of each database type.

Also, I envision that we could potentially have extra tables in a specific database (e.g. for `labels`) in case the JSON support is not as powerful/performant as the Postgres one, and in that case I believe having a library like `go-sqlbuilder` will make things more dificult.

In other words, I believe that we could force us to use a library like `go-sqlbuilder` but instead of making the things easier for readability/maintenance it would make things worse. 

PTAL at my PR (currently working) and if you still think that we should integrate our code with an already built sql builder library, I will create a separate issue to address it.

**Comment by ggallen** [comment]
> For example the go-sqlbuilder forces you to have the SELECT and FROM clauses separated, and it provides a lot of condition methods that I believe we don't need as we rely a lot on the specific JSON functionalities of each database type.
>
> Also, I envision that we could potentially have extra tables in a specific database (e.g. for labels) in case the JSON support is not as powerful/performant as the Postgres one, and in that case I believe having a library like go-sqlbuilder will make things more dificult.

The statements seem to contradict each other.  The SQL builder provides to many features? Doesn't saying that sound odd?

In the first paragraph you say that we use specific JSON functionality so we don't need these conditions. In the second you say we might at some point we might not have a database that provides that functionality, but in that case having those extra conditions will make it more difficult. For me that does not compute.

We are writing something here that has been done before. There are lots of SQL builder packages, and it's not clear to me why we think we can write a better one, nor why we want to.

> PTAL at my PR (currently working) and if you still think that we should integrate our code with an already built sql builder library, I will create a separate issue to address it.

I can, but I don't think this is the way we should be developing things here.  It seems like our strategy is often to implement something and then immediately open a PR to fix or refactor it.  For significant things like this we should investigate, have a discussion, and then decide on a path forward.
